### PR TITLE
migrate diff.diff to cmp.Diff

### PR DESCRIPTION
**Pull request by danielqsj**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

`diff.ObjectDiff`, `diff.ObjectGoPrintDiff` and `diff.ObjectReflectDiff` are deprecated from v1.15.
This PR aims to migrate existing code in kubernetes repo to use cmp.Diff instead of these funcs.

**Which issue(s) this PR fixes**:

Follow up with https://github.com/kubernetes/kubernetes/pull/70929

**Special notes for your reviewer**:

/assign @mikedanese

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by danielqsj** [comment]
no api changed
/remove-kind api-change

File: pkg/api/persistentvolume/util_test.go [hunk 1]
```diff
@@ -134,10 +134,10 @@ func TestDropDisabledFields(t *testing.T) {
 
 			DropDisabledFields(tc.newSpec, tc.oldSpec)
 			if !reflect.DeepEqual(tc.newSpec, tc.expectNewSpec) {
```
Comment (mikedanese) [inline pkg/api/persistentvolume/util_test.go]: Can you change `reflect.DeepEqual` to `cmp.Equal` to make sure these tests aren't depending on private field comparison?

Comment (danielqsj) [reply pkg/api/persistentvolume/util_test.go]: @mikedanese yep, for better review, I will create a new PR for this change.

Comment (danielqsj) [reply pkg/api/persistentvolume/util_test.go]: @mikedanese 
So this `apiequality.Semantic.DeepEqual` should also be replaced ? https://github.com/kubernetes/kubernetes/blob/4e63af2324b8562ac41f46a2e82b1b8a4224c048/pkg/kubelet/kubelet_node_status_test.go#L312

**Pull request closed without merging**